### PR TITLE
Add TypeScript (+ React) support

### PR DESCRIPTION
**Pull request by firstred**
Adds eslint configs for TypeScript and React as used in https://github.com/mollie/mollie-api-node and https://github.com/mollie/PrestaShop

### CODE REVIEW

**Comment by firstred** [comment]
NOTE (Feb 2019): this is how you currently have to activate TypeScript support in Webstorm/PhpStorm: https://intellij-support.jetbrains.com/hc/en-us/community/posts/115000225170-ESLint-and-ts-Typescript-files?page=1#community_comment_360000332879

**Comment by firstred** [comment]
Visual Studio Code:
Install the ESLint extension (`dbaeumer.vscode-eslint`).
Add
```json
"eslint.validate": [
    "javascript",
    "javascriptreact",
    { "language": "typescript", "autoFix": true },
    { "language": "typescriptreact", "autoFix": true }
]
```
to `settings.json` to enable TypeScript support + auto fix suggestions.

**Comment by firstred** [comment]
ESLint cli: `eslint --ext .js,.jsx,.ts,.tsx src/`

File: packages/eslint-config-react/package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "eslint": ">= 3"
   },
   "dependencies": {
-    "@mollie/eslint-config-base": "^0.0.1-alpha.2",
+    "@mollie/eslint-config-base": "^0.0.1-alpha.3",
```
Comment (vernondegoede) [inline packages/eslint-config-react/package.json]: Nice one 👍
Leave the other versions for now. I'll version bump all of them to the correct versions upon release (after merging).

File: packages/eslint-config-typescript/package.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    "access": "public"
+  },
+  "peerDependencies": {
+    "eslint": ">= 3"
+  },
+  "dependencies": {
+    "@mollie/eslint-config-base": "^0.0.1-alpha.3",
+    "@typescript-eslint/eslint-plugin": "^1.3.0",
+    "@typescript-eslint/parser": "^1.3.0",
+    "typescript": "^3.3.3"
```
Comment (vernondegoede) [inline packages/eslint-config-typescript/package.json]: Do we need this as a dependency? Or can it be a peerDepencency?

Comment (firstred) [reply packages/eslint-config-typescript/package.json]: It looks like a peerDependency of `typescript >= 3` will do. I'll change it.

Comment (firstred) [reply packages/eslint-config-typescript/package.json]: Nope, `typescript >= 3.2.1`. The ESLint plugin requires a relatively high version.

Comment (firstred) [reply packages/eslint-config-typescript/package.json]: And `eslint >= 5.12.1` it seems.

**Pull request merged**